app/scripts/lib/stores/read-only-network-store.ts

```diff
@@ -0,0 +1,89 @@
+import log from 'loglevel';
+import getFetchWithTimeout from '../../../../shared/modules/fetch-with-timeout';
+import {
+  type IntermediaryStateType,
+  BaseStore,
+  MetaMaskStorageStructure,
+} from './base-store';
+
+const fetchWithTimeout = getFetchWithTimeout();
+
+const FIXTURE_SERVER_HOST = 'localhost';
+const FIXTURE_SERVER_PORT = 12345;
+const FIXTURE_SERVER_URL = `http://${FIXTURE_SERVER_HOST}:${FIXTURE_SERVER_PORT}/state.json`;
+
+/**
+ * A read-only network-based storage wrapper
+ */
+export default class ReadOnlyNetworkStore extends BaseStore {
+  #initialized: boolean = false;
+
+  #initializing?: Promise<void>;
+
+  #state: IntermediaryStateType | null = null;
+
+  constructor() {
+    super();
+    this.#initializing = this.#init();
+  }
+
+  /**
+   * Declares this store as compatible with the current browser
+   */
+  isSupported = true;
+
+  /**
+   * Initializes by loading state from the network
+   */
+  async #init() {
+    try {
+      const response = await fetchWithTimeout(FIXTURE_SERVER_URL);
+
+      if (response.ok) {
+        this.#state = await response.json();
+      } else {
+        log.debug(
+          `Received response with a status of ${response.status} ${response.statusText}`,
+        );
+      }
+    } catch (error) {
+      console.log('error', error);
+      if (error instanceof Error) {
+        log.debug(`Error loading network state: '${error.message}'`);
+      } else {
+        log.debug(`Error loading network state: An unknown error occurred`);
+      }
+    } finally {
+      this.#initialized = true;
+    }
+  }
+
+  /**
+   * Returns state
+   */
+  async get() {
+    if (!this.#initialized) {
+      await this.#initializing;
+    }
+    return this.#state;
+  }
+
+  /**
+   * Overwrite in-memory copy of state.
+   *
+   * @param obj - The data to set
+   * @param obj.data - The MetaMask State tree
+   * @param obj.meta - The metadata object
+   * @param obj.meta.version - The version of the state tree determined by the
+   * migration
+   */
+  async set(obj: MetaMaskStorageStructure): Promise<void> {
+    if (!obj) {
+      throw new Error('MetaMask - updated state is missing');
+    }
+    if (!this.#initialized) {
+      await this.#initializing;
+    }
+    this.#state = obj;
```

Comment:
Nit: `obj` is a poor description of what this parameter is. I chose `data` here to match the JSDoc param description, though something like `state` might be good too.

Also the state property descriptions here seem redundant, those are in the type already.

```suggestion
   * @param data - The data to set
   */
  async set(data: MetaMaskStorageStructure): Promise<void> {
    if (!data) {
      throw new Error('MetaMask - updated state is missing');
    }
    if (!this.#initialized) {
      await this.#initializing;
    }
    this.#state = data;
```

Reply:
suggestion applied in 3b5e8a7b25